pkglistgen/tool.py

```diff
@@ -696,17 +696,24 @@ def update_and_solve_target(
                 logging.info(f'{project}/{product} build in progress')
                 return
         if git_url:
+            git_url_base, *fragment = git_url.split('#')
             if os.path.exists(cache_dir + "/.git"):
                 # reset and update existing clone
                 logging.debug(subprocess.check_output(
                     ['git', 'reset', '--hard'], cwd=cache_dir, encoding='utf-8'))
                 logging.debug(subprocess.check_output(
                     ['git', 'pull', '--rebase'], cwd=cache_dir, encoding='utf-8'))
+                if fragment:
+                    # FIXME: this is not takeing account default branches, which may change server side
+                    logging.debug(subprocess.check_output(
+                        ['git', 'switch', fragment], cwd=cache_dir, encoding='utf-8'))
```

Comment:
`git switch` is experimental and might not be available on ibs-slsa-botmaster

Reply:
the git on botmaster knows git switch

Reply:
also git on ibs-slsa-botmaster knows it